UnicornEngine/video/source/vulkan/Image.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/vulkan/Image.hpp>
+#include <unicorn/utility/Logger.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+namespace vulkan
+{
+Image::Image(): m_format()
```

Comment:
Image instance is not initialized in the constructor, it does not guaranty RAII, and it does not hold any member providing the info about whether it is initialized or not. So it's not possible to **explicitly** check initialization state in the user code.
Also, it's being a class managing resources does not provide overloaded or omitted copy constructor and assignment operator, which might lead to unexpected behavior in the user code.